### PR TITLE
fix(#269): constraints which are cleared by clicking next 

### PR DESCRIPTION
**Pull request by kennsippell**
#269

### CODE REVIEW

**Comment by kennsippell** [comment]
Hey @jkuester. Interested in your thoughts on this fix.

This test is passing on harness 3.x and failing after the form-filler changes when introducing cht-form. Specifically it is bailing [here](https://github.com/medic/cht-conf-test-harness/commit/9b3043f48fb5a8d2f262769d9ecbf8a06a17a79c#diff-a4fd2a8bb5172bdeac48273863aac8bc91fcb66c72bb35321d645433fa42c864R293).

Interested in your thoughts on the macro approach here.

Basically, the condition of the failure is that I'm on the last page of the form so there is no next button + I have this weird constraint which only clears when clicking Next.  

Clicking "submit" causes the test to hang presently because the promise just below never resolves. This was a really quick fix and I'm down to think this through more, but I wanted to engage your expertise in case you have quick thoughts.

**Comment by jkuester** [comment]
Sorry! I was not able to get this looked at this week.  I plan to do it first thing next week! :+1:

**Review by jkuester — APPROVED** [review]
Okay, thank you for the repro forms! Very easy to understand the behavior with those.  :+1:   Good news and bad news:

Bad news is that it seems pretty clear that this is caused by https://github.com/enketo/enketo/issues/62.  I have not been able to identify any viable workaround that could be implemented in a form.  This really needs to be fixed in Enketo (or if that turns out to be impossible, we could cook something up in cht-core....).  FTR I did spend some time trying to tinker with the technically unsupported `trigger` column to see if I could get that to work, but I could not get any useful behavior there.

The good news is that the "workaround" of just clicking Next or Submit is viable (for both cases) and should behave as desired.  Any stale constraint violations should be cleared and the current value for the question will be recorded.  When advancing to the next page, Enketo will [trigger validation](https://github.com/enketo/enketo-core/blob/e80306a5db69125bab9e8f14085eed9155494dfe/src/js/page.js#L367) to ensure there are no violated constraints.  Then, when Submitting the form, the cht-core code will [re-validate the whole form](https://github.com/medic/cht-core/blob/282faee191a448bbcff8f67baa8b3fc844ef14dc/webapp/src/ts/services/enketo.service.ts#L551) to ensure everything is good before actually saving the report.

**Comment by kennsippell** [comment]
Thanks @jkuester. Appreciate you checking on this in-depth. Let's continue to work with Enketo for an underlying fix.
